File: backend/src/main/java/com/example/gestion_academica/service/TemaService.java
package com.example.gestion_academica.service;

import java.util.List;
import java.util.Optional;

import org.springframework.http.ResponseEntity;

import com.example.gestion_academica.entity.Tema;


public interface TemaService {

	List<Tema> obtenerTodosLosTemas();

	Optional<Tema> obtenerTemaPorId(Long id);

	ResponseEntity<Tema> crearTema(Tema tema);

	boolean eliminarTema(Long id);

	ResponseEntity<Tema> actualizarTema(Long id, Tema updatedTema);

	

	

}

File: frontend/src/app/services/tema.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Tema } from '../models/tema.model'; // Importa el modelo Tema

@Injectable({
  providedIn: 'root',
})
export class TemaService {
  private apiUrl = 'http://localhost:8080/tema'; // URL de tu API

  constructor(private http: HttpClient) {}

  // Obtener todos los temas
  obtenerTodosLosTemas(): Observable<Tema[]> {
    return this.http.get<Tema[]>(this.apiUrl);
  }

  // Obtener un tema por ID
  obtenerTemaPorId(id: number): Observable<Tema> {
    return this.http.get<Tema>(`${this.apiUrl}/${id}`);
  }

  // Crear un nuevo tema
  crearTema(tema: Tema): Observable<Tema> {
    return this.http.post<Tema>(this.apiUrl, tema);
  }

  // Actualizar un tema existente
  actualizarTema(id: number, tema: Tema): Observable<Tema> {
    return this.http.put<Tema>(`${this.apiUrl}/${id}`, tema);
  }

  // Eliminar un tema por ID
  eliminarTema(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
